common/lib/xmodule/xmodule/tabs.py

```diff
@@ -400,12 +401,16 @@ def initialize_default(course):
         else:
             discussion_tab = CourseTab.load('discussion')
 
-        course.tabs.extend([
+        course_tabs.extend([
             CourseTab.load('textbooks'),
             discussion_tab,
             CourseTab.load('wiki'),
             CourseTab.load('progress'),
         ])
+        # While you should be able to do `tab.priority`, a lot of tests mock tabs to be a dict
+        # which causes them to throw an error on this line
+        course_tabs.sort(key=lambda tab: getattr(tab, 'priority', None) or float('inf'))
```

Comment:
1. Can you link some of these tests?
2. When you load a tab using `.load(....)` does it not return a dict?

Reply:
1. Here is an example file that has a lot of statements patching the `CourseTab.from_json` method, which is used by `CourseTab.load`: https://github.com/edx/edx-platform/blob/75919473f3c6a21b0ada05099c4ec22235fdc8ed/common/lib/xmodule/xmodule/modulestore/tests/test_split_modulestore.py#L585
2. No. In the actual platform it will load the CourseTab plugin and return an instance of the plugin with the tab data passed to it. However, in the tests it just returns a dict (I guess to avoid involving plugins which might complicate test code).

Reply:
Got it thanks, follow up question - when the tests expect a dict - is there a mock in the test that handles this scenario?

Reply:
By this scenario, do you mean testing with tab priorities? I think that is being tested by the tab reordering code since it will rely on tab priorities to keep tabs in order. If the priorities change, the tests will fail since reodering will cause the order to change.